KarhooUISDK/Extensions/UIColor+extension.swift

```diff
@@ -16,29 +16,56 @@ extension UIColor {
                        alpha: 1.0)
     }
     
-    public convenience init?(hex: String) {
-        let r, g, b, a: CGFloat
+    public convenience init(hex: String, alpha: CGFloat = 1) {
+        var hex = hex
 
-        if hex.hasPrefix("#") {
-            let start = hex.index(hex.startIndex, offsetBy: 1)
-            let hexColor = String(hex[start...])
+        if !hex.hasPrefix("#") || hex.count != 7 {
+            assertionFailure("# + six digits color code expected to decode hex value")
+            UIColor.adjustInvalidHex(&hex)
+        }
 
-            if hexColor.count == 8 {
-                let scanner = Scanner(string: hexColor)
-                var hexNumber: UInt64 = 0
+        let scanner = Scanner(string: hex)
+        scanner.scanLocation = 1  // skip #
+        
+        var rgb: UInt32 = 0
+        scanner.scanHexInt32(&rgb)
+        
+        self.init(
+            red: CGFloat((rgb & 0xFF0000) >> 16)/255.0,
+            green: CGFloat((rgb &   0xFF00) >>  8)/255.0,
+            blue: CGFloat((rgb &     0xFF)      )/255.0,
+            alpha: alpha
+        )
+    }
 
-                if scanner.scanHexInt64(&hexNumber) {
-                    r = CGFloat((hexNumber & 0xff000000) >> 24) / 255
-                    g = CGFloat((hexNumber & 0x00ff0000) >> 16) / 255
-                    b = CGFloat((hexNumber & 0x0000ff00) >> 8) / 255
-                    a = CGFloat(hexNumber & 0x000000ff) / 255
+    private static func adjustInvalidHex(_ hex: inout String) {
+        if !hex.hasPrefix("#") {
+            assertionFailure("# expected at the begining of hex value")
+            hex = "#" + hex
+        }
 
-                    self.init(red: r, green: g, blue: b, alpha: a)
-                    return
-                }
+        if hex.count != 7 {
+            assertionFailure("Six digits color code expected expected")
+            switch hex.count {
+            case ..<7:
+                hex += (1 ..< 7 - hex.count).map { _ in "0" }.joined()
+            case 7:
+                break
+            case 7...:
+                hex.removeLast(7 - hex.count)
+            default:
+                assertionFailure()
             }
         }
+    }
 
-        return nil
+    static func get(lightModeColor: UIColor, darkModeColor: UIColor) -> UIColor {
+        if #available(iOS 13, *) {
+            return UIColor { (traitCollection) -> UIColor in
+                return traitCollection.userInterfaceStyle == .light ? lightModeColor : darkModeColor
```

Comment:
There's also .unspecified. Could you check with Olivier what's the default in that case? We may need to change the if to `if style == .dark ? darkModeColor : lightModeColor` (depending on his answer)